### PR TITLE
Updated the show-action command to include the default value of action parameters

### PR DESCRIPTION
**Pull request by jnsgruk**
This pull request adds the ability for the `juju show-action` command to also display the default value for any parameters where they are specified/non-empty.

## QA steps

```sh
$ juju bootstrap microk8s micro
$ juju deploy jnsgruk-test
$ juju show-action jnsgruk-test example-action
An example action with some parameters.

Arguments
param-blank-default:
  type: string
  description: this parameter has a blank default value.
  default: ""
param-default-value:
  type: string
  description: this parameter has a default value.
  default: somevalue
param-no-default:
  type: string
  description: this parameter has no default.
```

Note that the `jnsgruk-test` charm essentially does nothing but [specify some actions](https://github.com/jnsgruk/test-charm/blob/show-action-test/actions.yaml) and start a blank ubuntu container.
## Bug reference

https://bugs.launchpad.net/juju/+bug/1908714

### CODE REVIEW

**Comment by jameinel** [comment]
!!build!!

File: cmd/juju/action/show.go [hunk 1]
```diff
@@ -102,10 +102,21 @@ func (c *showCommand) Run(ctx *cmd.Context) error {
 			if !ok {
 				continue
 			}
-			args[argName] = actionArg{
-				Type:        infoMap["type"],
-				Description: infoMap["description"],
+
+			if infoMap["default"] != nil {
```
Comment (hmlanigan) [inline cmd/juju/action/show.go:106]: You can check that the key exists and get the value if it exists in one step with something like:

			action := actionArg{
				Type:        infoMap["type"],
				Description: infoMap["description"],
			}
			if value, ok := infoMap["default"]; ok {
				action.Default = value
			}
			args[argName] = action

File: cmd/juju/action/show.go [hunk 2]
```diff
@@ -102,10 +102,21 @@ func (c *showCommand) Run(ctx *cmd.Context) error {
...
+				args[argName] = actionArg{
+					Type:        infoMap["type"],
+					Description: infoMap["description"],
+					Default:     infoMap["default"],
```
Comment (jameinel) [inline cmd/juju/action/show.go:110]: setting an interface{} to 'nil' is the same as not setting it, so I think you can just have
`args[argName] = actionArg{Type=,Description=,Default=}`

without doing any other checking.

File: cmd/juju/action/show_test.go [hunk 3]
```diff
@@ -87,9 +87,14 @@ Arguments
...
   type: boolean
   description: take a full backup
+  default: true
 name:
   type: string
   description: snapshot name
+prefix:
+  type: string
+  description: prefix to snapshot name
+  default: ""
```
Comment (jameinel) [inline cmd/juju/action/show_test.go:97]: I think we are now missing the case where there is no default, are we?

File: cmd/juju/action/show.go (hunk 1 above)
Comment (jnsgruk) [reply cmd/juju/action/show.go:106]: Fixed per @jameinel comment in the next commit, thanks!

File: cmd/juju/action/show.go (hunk 2 above)
Comment (jnsgruk) [reply cmd/juju/action/show.go:110]: Much nicer, fixed. Thanks!

File: cmd/juju/action/show_test.go (hunk 3 above)
Comment (jnsgruk) [reply cmd/juju/action/show_test.go:97]: I adjusted the [test actions](https://github.com/juju/juju/blob/1b2093b01b91e6020ac0084c2f6030bf194daf5c/cmd/juju/action/package_test.go#L80) in `package_test.go` so I think we're covered. We have a test for no default, a blank default and a default value

Comment (jnsgruk) [reply cmd/juju/action/show_test.go:97]: See the "name" field above? :)

**Review by hmlanigan — APPROVED** [review]
QA is good.  Changes and tests LGTM.

**Comment by hmlanigan** [comment]
!!build!!

File: cmd/juju/action/show_test.go (hunk 3 above)
Comment (jameinel) [reply cmd/juju/action/show_test.go:97]: yep, sorry I missed it.

**Review by jameinel — APPROVED** [review]

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**